File: binary-number-with-alternating-bits.ts
{
    function hasAlternatingBits(n: number): boolean {
        let str = n.toString(2);

        if (str.length <= 2) 
            str += '1';

        for (let i = 1; i < str.length - 1; i++) {
            if (str[i - 1] == str[i] || str[i + 1] == str[i])
                return false
        }
        return true;
    };

    console.log(hasAlternatingBits(2))
}

File: binary-number-with-alternating-bits.rs
pub fn has_alternating_bits(mut n: i32) -> bool {
    let mut prev = -1;
    while n > 0 {
        if prev == n % 2 {
            return false;
        }
        prev = n % 2;
        n /= 2;
    }
    true
}
